src/System.Reflection.Metadata/src/System/Reflection/Metadata/Internal/MetadataFlags.cs

```diff
@@ -120,86 +122,194 @@ internal enum HeapSizeFlag : byte
+
+        // Namespace handle for namespace with child namespaces but no types of its own           
+        internal const uint SyntheticNamespace = Namespace | (1 << OffsetBitCount);
+
+        // Reserved namespaces that can be used in future:
+        internal const uint ReservedNamespace1 = Namespace | (2 << OffsetBitCount);
+        internal const uint ReservedNamespace2 = Namespace | (3 << OffsetBitCount);
+
+        // Reserved virtual namespaces that can be used in future:
+        internal const uint ReservedVirtualNamespace1 = VirtualBit | Namespace;
+        internal const uint ReservedVirtualNamespace2 = VirtualBit | Namespace | (1 << OffsetBitCount);
+        internal const uint ReservedVirtualNamespace3 = VirtualBit | Namespace | (2 << OffsetBitCount);
+        internal const uint ReservedVirtualNamespace4 = VirtualBit | Namespace | (3 << OffsetBitCount);
+
+        // Heap offset values are limited to 29 bits (max compressed integer)
+        internal const int OffsetBitCount = 29;
+        internal const uint OffsetMask = (1 << OffsetBitCount) - 1;
+        internal const uint NonVirtualTypeMask = 3u << OffsetBitCount;
+        internal const uint TypeMask = VirtualBit | NonVirtualTypeMask;
+    }
+
+    internal static class HandleType
     {
-        internal const uint Module = 0x00000000;
-        internal const uint TypeRef = 0x01000000;
-        internal const uint TypeDef = 0x02000000;
-        internal const uint FieldDef = 0x04000000;
-        internal const uint MethodDef = 0x06000000;
-        internal const uint ParamDef = 0x08000000;
-        internal const uint InterfaceImpl = 0x09000000;
-        internal const uint MemberRef = 0x0a000000;
-        internal const uint Constant = 0x0b000000;
-        internal const uint CustomAttribute = 0x0c000000;
-        internal const uint DeclSecurity = 0x0e000000;
-        internal const uint Signature = 0x11000000;
-        internal const uint EventMap = 0x12000000;
-        internal const uint Event = 0x14000000;
-        internal const uint PropertyMap = 0x15000000;
-        internal const uint Property = 0x17000000;
-        internal const uint MethodSemantics = 0x18000000;
-        internal const uint MethodImpl = 0x19000000;
-        internal const uint ModuleRef = 0x1a000000;
-        internal const uint TypeSpec = 0x1b000000;
-        internal const uint Assembly = 0x20000000;
-        internal const uint AssemblyRef = 0x23000000;
-        internal const uint File = 0x26000000;
-        internal const uint ExportedType = 0x27000000;
-        internal const uint ManifestResource = 0x28000000;
-        internal const uint NestedClass = 0x29000000;
-        internal const uint GenericParam = 0x2a000000;
-        internal const uint MethodSpec = 0x2b000000;
-        internal const uint GenericParamConstraint = 0x2c000000;
-
-        internal const uint UserString = 0x70000000;     // #UserString heap
+        internal const uint Module = 0x00;
```

Comment:
Sometimes they are useful as uints (for bitwise operations). ```tokenTypeSmall``` constants are there so that the methods that use them don't differ among handles.